pandas/core/internals/blocks.py

```diff
@@ -1388,12 +1388,13 @@ def interpolate(
             # If there are no NAs, then interpolate is a no-op
             return [self.copy(deep=False)]
 
-        # TODO(3.0): this case will not be reachable once GH#53638 is enforced
         if self.dtype == _dtype_obj:
             # only deal with floats
             # bc we already checked that can_hold_na, we don't have int dtype here
             # test_interp_basic checks that we make a copy here
-            return [self.copy(deep=False)]
+            raise TypeError(
+                f"{type(self).__name__} cannot interpolate with object dtype."
```

Comment:
we dont need to have Block in the exception message. the comments in the line above can be removed.  can you add "# GH#53631"

Reply:
can do something like `name = {1: "Series", 2: "DataFrame"}[self.ndim]` to keep the exception message unchanged

Reply:
thank you, I made changes, ci - green.